intake_bluesky/core.py

```diff
@@ -452,10 +548,12 @@ def read_partition(self, i):
         descriptor_uids = [doc['uid'] for doc in self._descriptors]
         skip = max(0, start - len(payload))
         limit = stop - start - len(payload)
-        # print('start, stop, skip, limit', start, stop, skip, limit)
         if limit > 0:
-            events = self._get_event_cursor(descriptor_uids=descriptor_uids,
-                                            skip=skip, limit=limit)
+
+            events = itertools.islice(interlace_event_pages(
+                    *[self._get_event_pages(descriptor_uid=descriptor_uid)
```

Comment:
Using `(` in place of `[` will make this a _generator comprehension_ instead of a _list comprehension_ and save one iteration.